api/producer/updateDocumentReference/tests/test_update_document_reference.py

```diff
@@ -821,8 +836,6 @@ def test_update_document_reference_existing_invalid_json(
     doc_pointer.document = "invalid json"
     repository.create(doc_pointer)
 
-    doc_ref.status = "draft"
```

Comment:
it could be worth adding a few unit tests for the new behaviour, making sure the status being "current" or not is behaving as we want for create, upsert and update handlers.